src/pubtools/_impl/tracing.py

```diff
@@ -12,14 +12,23 @@ def func():
 import os
 import threading
 
-from opentelemetry import baggage, context, trace
-from opentelemetry.baggage.propagation import W3CBaggagePropagator
-from opentelemetry.propagate import set_global_textmap
-from opentelemetry.sdk.resources import SERVICE_NAME, Resource
-from opentelemetry.sdk.trace import TracerProvider
-from opentelemetry.sdk.trace.export import BatchSpanProcessor, ConsoleSpanExporter
-from opentelemetry.trace import Status, StatusCode
-from opentelemetry.trace.propagation.tracecontext import TraceContextTextMapPropagator
+try:
+    from opentelemetry import baggage, context, trace
+    from opentelemetry.baggage.propagation import W3CBaggagePropagator
+    from opentelemetry.propagate import set_global_textmap
+    from opentelemetry.sdk.resources import SERVICE_NAME, Resource
+    from opentelemetry.sdk.trace import TracerProvider
+    from opentelemetry.sdk.trace.export import BatchSpanProcessor, ConsoleSpanExporter
+    from opentelemetry.trace import Status, StatusCode
+    from opentelemetry.trace.propagation.tracecontext import (
+        TraceContextTextMapPropagator,
+    )
+
+    OPENTELEMETRY_AVAILABLE = True
+except ImportError:
+    # Clients aren't expected to have open-telemetry. This flag will be used in
+    # TracingWrapper to provide pass through functions.
+    OPENTELEMETRY_AVAILABLE = False
```

Comment:
This could have a `pragma: no cover` added to pass the coverage check, given that it's probably not worth maintaining two separate test environments with otel present vs missing.